internal/nodes/avi_model_nodes.go

```diff
@@ -1302,7 +1324,9 @@ func (v *AviHTTPDataScriptNode) GetCheckSum() uint32 {
 func (v *AviHTTPDataScriptNode) CalculateCheckSum() {
 	// A sum of fields for this VS.
 	checksum := lib.DSChecksum(v.PoolGroupRefs, nil, false)
-	checksum = utils.Hash(fmt.Sprint(checksum) + utils.HTTP_DS_SCRIPT_MODIFIED)
+	for _, pgRefName := range v.PoolGroupRefs {
```

Comment:
I think DS will point to single PG. Do we need this for loop.?

Reply:
yes, you are right. but some UTs panicked when I accessed it with the slice with 0th index.

Reply:
I think we should look into panic trace into depth. It could be somewhere we are missing setting up things properly.

Reply:
it is failing at the place where we are calculating the checksum of DS after dissociation of PG from DS. I will add a condition check whether the number of PGs associated is 1, if yes, include the DS.